src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -787,6 +796,8 @@ export class HubConnection {
             this._connectionState = HubConnectionState.Disconnected;
             this._connectionStarted = false;
 
+            this._stopPromise = undefined;
```

Comment:
I'm not saying this is a bad idea, but is there any reason this is actually necessary? We should never use the `_stopPromise` unless we know the current connection is currently in the process of disconnecting.